shell/platform/fuchsia/dart-pkg/zircon/lib/src/system.dart

```diff
@@ -63,23 +62,31 @@ class HandlePairResult extends _Result {
 
 @pragma('vm:entry-point')
 class ReadResult extends _Result {
-  final ByteData bytes;
-  final int numBytes;
-  final List<Handle> handles;
+  final ByteData? bytes;
+  final int? numBytes;
+  final List<Handle>? handles;
   @pragma('vm:entry-point')
   const ReadResult(final int status, [this.bytes, this.numBytes, this.handles])
       : super(status);
-  Uint8List bytesAsUint8List() =>
-      bytes.buffer.asUint8List(bytes.offsetInBytes, numBytes);
-  String bytesAsUTF8String() => utf8.decode(bytesAsUint8List());
+
+  Uint8List? bytesAsUint8List() {
+    if (bytes == null) {
+      return null;
+    }
+    return bytes!.buffer.asUint8List(bytes!.offsetInBytes, numBytes!);
+  }
+
+
+  String? bytesAsUTF8String() => bytes == null ? null : utf8.decode(bytesAsUint8List()!);
```

Comment:
Ditto - consider returning non-nullable String and removing the test for null.

Reply:
done.